.github/workflows/r-valgrind.yaml

```diff
@@ -14,20 +14,45 @@ jobs:
       image: rocker/r2u:latest
     steps:
       - uses: actions/checkout@v3
+
       - name: SessionInfo
         run: R -q -e 'sessionInfo()'
+
       - name: System Dependencies
         run: apt update -qq && apt upgrade --yes && apt install --yes --no-install-recommends valgrind cmake git
+
+      - name: Set additional repositories (Linux)
+        run: |
+          rversion <- paste(strsplit(as.character(getRversion()), split = '\\.')[[1L]][1:2], collapse = '.')
+          codename <-  system('. /etc/os-release; echo ${VERSION_CODENAME}', intern = TRUE)
+          repo <- "https://tiledb-inc.r-universe.dev"
+          (opt <- sprintf('options(repos = c("%s/bin/linux/%s/%s", "%s", getOption("repos")), timeout = 300L)', repo, codename, rversion, repo))
+          print("RVERSION")
+          print(rversion)
+          print("CODENAME")
+          print(codename)
+          print("REPO")
+          print(repo)
+          cat(opt, "\n", file = "~/.Rprofile", append = TRUE)
+        shell: Rscript {0}
+
+      - name: Install tiledb-r
+        run: cd apis/r && Rscript tools/install-tiledb-r.R
+
       - name: Package Dependencies
-        run: cd apis/r && R -q -e 'remotes::install_deps(".", dependencies=TRUE)'
+        run: cd apis/r && R -q -e 'remotes::install_deps(".", dependencies=TRUE, upgrade=FALSE)'
+
       - name: Fetch tiledb from r-universe
         ## DESCRIPTION has a 'hard' depends on a tiledb version not at CRAN and hence not in r2u
         ## So using a helper script from the `littler` package to fetch the Ubuntu binary from the given universe
         run: cd apis/r && installRub.r -u eddelbuettel tiledb
+
```

Comment:
Is this bit necessary? If you're using the `tools/install-tiledb-r.R` script, then we shouldn't need this "controlled downgrade"

Reply:
@mojaveazure I don't know, I'm just attempting to imitate your prior work with your guidance.

I still have low R expertise and am functioning in a clerical role here. Open to feedback.

Reply:
Commit #6: https://github.com/single-cell-data/TileDB-SOMA/pull/2711/commits/8e3350c91903ed55a921f8c1d860c8a275ec82fe

Reply:
I didn't write this file, so this is not my prior work. What I did for `r-ci.yml` was
- add R-universe as an additional repository
- add a script `tools/install-tiledb-r.R` to install the correct version of tiledb-r based on the constraints from `DESCRIPTION`
- force remotes to not upgrade dependencies

This obviates the need for a controlled downgrade as `tools/install-tiledb-r.R` installs the correct version of tiledb-r from CRAN or R-universe and remotes no-longer greedily upgrades dependencies

Reply:
Manual CI run for commit #6: https://github.com/single-cell-data/TileDB-SOMA/actions/runs/9471423859

Reply:
The script I wrote worked (under a different version "skew").  I lost track of what works where now.  All I can tell is the nightly is now broken and that is not as intended:

![image](https://github.com/single-cell-data/TileDB-SOMA/assets/673121/5276f48d-0195-42b2-8a18-f6d77531b848)

Reply:
> I didn't write this file, so this is not my prior work

I am attempting to imitate your prior work in #2540 on other CI files, onto this CI file

* #2540 touched CI files that run on every PR
* #2703 was my attempt to imitate your #2540 work an R CI YAML file that doesn't run on every PR
* This PR is my attempt to imitate your #2540 on yet another R CI YAML file that doesn't run on every PR

Reply:
> All I can tell is the nightly is now broken and that is not as intended:

Correct, that's why I'm working on this PR.

Reply:
> Manual CI run for commit https://github.com/single-cell-data/TileDB-SOMA/pull/6: https://github.com/single-cell-data/TileDB-SOMA/actions/runs/9471423859

This is ☘️